### PR TITLE
cleanup: make CHANGED_FILES_PATH mandatory when building a run plan

### PR DESCRIPTION
**Pull request by ptab**

### CODE REVIEW

File: .github/workflows/validate-pr-python.yaml [hunk 1]
```diff
@@ -93,6 +93,13 @@ jobs:
       - name: Test
         run: pipenv run test-ci
 
+      - uses: mikepenz/action-junit-report@v4.3.1
```
Comment (ptab) [inline .github/workflows/validate-pr-python.yaml:96]: Boyscout rule: report these test results to the run summary so that we don't have to dive into the logs every time.

File: src/mpyl/steps/run_properties.py [hunk 2]
```diff
@@ -16,53 +15,42 @@ def construct_run_properties(
     config: dict,
     properties: dict,
     cli_parameters: MpylCliParameters = MpylCliParameters(),
-    run_plan: Optional[RunPlan] = None,
```
Comment (ptab) [inline src/mpyl/steps/run_properties.py:19]: These two parameters are only ever used for test purposes, and add a lot of complexity to the original function. These were replaced by a new [`stub_run_properties()` test function](https://github.com/Vandebron/gh-mpyl/pull/38/files#diff-98e5a744692c47a2036490c9579b543bcd95c23f71a5250db07de7cf041f60c8R22-R37).

File: src/mpyl/stages/discovery.py [hunk 3]
```diff
@@ -283,19 +282,21 @@ def create_run_plan(
...
-        logger=logger,
-        repository=repository,
-        all_projects=all_projects,
-        all_stages=all_stages,
-        selected_projects=selected_projects,
-        selected_stage=selected_stage,
-        tag=tag,
-        changed_files_path=changed_files_path,
-    )
+    if changed_files_path:
```
Comment (Jorg88) [inline src/mpyl/stages/discovery.py:285]: This seems weird considering the pr title 😬 why is this needed? You can make the field required in the schema so the config won't even load if it's not set

File: src/mpyl/steps/run_properties.py [hunk 4]
```diff
@@ -16,53 +15,42 @@ def construct_run_properties(
...
-                    explain_run_plan=explain_run_plan,
-                    repo=repo,
-                    tag=tag,
-                    changed_files_path=changed_files_path,
-                )
+        stages = [Stage(stage["name"], stage["icon"]) for stage in properties["stages"]]
+        run_plan_logger = logging.getLogger("mpyl")
+        if explain_run_plan:
+            run_plan_logger.setLevel("DEBUG")
+        changed_files_path = config["vcs"].get("changedFilesPath", None)
```
Comment (Jorg88) [inline src/mpyl/steps/run_properties.py:38]: and otherwise (if you don't want to change the schema) I think it's better to check for it right after loading it here

Comment (ptab) [reply src/mpyl/steps/run_properties.py:38]: That's exactly what I wanted to do, but unfortunately we can't because we only check for an existing run plan in disk 10 stack calls later, meaning we only know if we need this file or not much later in the process 😢

I'll refactor this in a follow-up PR.

**Review by Jorg88 — APPROVED** [review]
